packages/react-cosmos-component-playground/src/components/FixtureList/filter.js

```diff
@@ -0,0 +1,38 @@
+// Adapted from https://github.com/alexcurtis/react-treebeard
+import { match } from 'fuzzaldrin-plus';
+
+// Helper functions for search filtering
+export const defaultMatcher = (filterText, node) => {
+  const matchText = node.urlParams
+    ? `${node.urlParams.component}${node.urlParams.fixture}`
+    : node.name;
+  return match(matchText, filterText).length > 0;
+};
+
+export const findNode = (node, filter, matcher) => {
+  return (
+    matcher(filter, node) || // i match
+    (node.children && // or i have decendents and one of them match
+      node.children.length &&
+      Boolean(node.children.find(child => findNode(child, filter, matcher))))
+  );
+};
+
+export const filterTree = (node, filter, matcher = defaultMatcher) => {
+  // The root object can be an array
+  if (node.constructor === Array) {
```

Comment:
Done, except we now have an XO issue due to circular function calls

Reply:
I guess you don't want a global 'no-use-before-define' rule override, so I added `// eslint-disable-next-line no-use-before-define`. Is there an XO specific syntax I should use instead?

Reply:
> Done, except we now have an XO issue due to circular function calls

This is because `const foo =` is a var declaration, and the right way to do it is to declare `function foo` for root level functions in a module. I've not be consistent with this myself as it's nicer to use arrow and to always write functions in a single way, but I'm planning to use `function` declarations for module exports from now on. So make them function declarations if it's not a big deal, otherwise I can live with since I'm sure there are plenty other similar cases in the codebase.

> I guess you don't want a global 'no-use-before-define' rule override, so I added // eslint-disable-next-line no-use-before-define. Is there an XO specific syntax I should use instead?

Nope, this is fine.